### PR TITLE
Upgrade curl to 8.5.0

### PR DESCRIPTION
**Pull request by yifeikong**

### CODE REVIEW

**Comment by bjia56** [comment]
NSS was removed in https://github.com/curl/curl/commit/7c8bae0d9c9b2dfeeb008b9a316117d7b9675175

We can see if reverting that commit in the upstream curl would allow NSS to be linked in again for Firefox, but unfortunately this may not be a long-term sustainable approach.

**Comment by yifeikong** [comment]
That's a pretty big commit, I wonder if we would be able to revert it.

**Comment by yifeikong** [comment]
Firefox starts to support ECH in recent versions, enabling that with NSS make things even worse.

**Comment by bjia56** [comment]
So it sounds like we're out of luck with the Firefox build and need to emulate using BoringSSL starting curl 8.5?

**Comment by yifeikong** [comment]
I'm afraid so, I still need to update the extensions and it should be done. I will first implement firefox120, other older versions should be similar. But if there are some ciphers that are too old to be added back, it won't be supported then.

File: README.md [hunk 1]
```diff
@@ -47,13 +47,6 @@ The following browsers can be impersonated.
...
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 95 | 95.0.2 | Windows 10 | `ff95` | [curl_ff95](firefox/curl_ff95) |
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 98 | 98.0 | Windows 10 | `ff98` | [curl_ff98](firefox/curl_ff98) |
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 100 | 100.0 | Windows 10 | `ff100` | [curl_ff100](firefox/curl_ff100) |
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 102 | 102.0 | Windows 10 | `ff102` | [curl_ff102](firefox/curl_ff102) |
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 109 | 109.0 | Windows 10 | `ff109` | [curl_ff109](firefox/curl_ff109) |
-| ![Firefox](https://raw.githubusercontent.com/alrra/browser-logos/main/src/firefox/firefox_24x24.png "Firefox") | 117 | 117.0.1 | Windows 10 | `ff117` | [curl_ff117](firefox/curl_ff117) |
```
Comment (T-256) [inline README.md:56]: Do u mind to create a tracking issue for bringing back old Firefox versions via customized fingerprint? (is it even possible?)

**Review by T-256 — APPROVED** [review]
That's huge! Thank you <3

**Comment by T-256** [comment]
fyi I manually put `libcurl.dll` built via [this release](https://github.com/T-256/curl-impersonate/releases/tag/v0.6.3) into curl_cffi.
It crashed on import on Python3.11

File: README.md (hunk 1 above)
Comment (yifeikong) [reply README.md:56]: Considering the current update strategy of Firefox, I think there versions would die out by the time when I implemented Firefox-only extensions with boringssl, so there is no plan to support them.

**Pull request merged**